clippy_lints/src/methods/contains_for_slice.rs

```diff
@@ -0,0 +1,42 @@
+use clippy_utils::diagnostics::span_lint;
+use rustc_hir::{Expr, ExprKind};
+use rustc_lint::LateContext;
+use rustc_middle::ty::{self};
+
+use super::{CONTAINS_FOR_SLICE, method_call};
+
+pub(super) fn check(cx: &LateContext<'_>, expr: &Expr<'_>) {
+    if !expr.span.from_expansion()
+    // any()
+    && let Some((name, recv, args, _, _)) = method_call(expr)
+    && name == "any"
+    // check if the inner closure is a equality check
+    && args.len() == 1
+    && let ExprKind::Closure(closure) = args[0].kind
+    && let body = cx.tcx.hir().body(closure.body)
+    && let ExprKind::Binary(op, _, _) = body.value.kind
+    && op.node == rustc_ast::ast::BinOpKind::Eq
+    // iter()
+    && let Some((name, recv, _, _, _)) = method_call(recv)
+    && name == "iter"
+    {
+        // check if the receiver is a u8/i8 slice
+        let ref_type = cx.typeck_results().expr_ty(recv);
+
+        match ref_type.kind() {
+            ty::Ref(_, inner_type, _)
+                if inner_type.is_slice()
+                    && let ty::Slice(slice_type) = inner_type.kind()
+                    && (slice_type.to_string() == "u8" || slice_type.to_string() == "i8") =>
```

Comment:
issue: we shouldn't be converting to a string here

Reply:
also I think it's fine for us to just universally have this lint? It's faster for slices, but it's still _cleaner_ for other things

Reply:
> we shouldn't be converting to a string here

I'm newbie so I don't know, but is the best practice here to use the `sym` macro?
https://github.com/rust-lang/rust-clippy/blob/f2aed50873e8e908119044c283c33cf6ef0b66e0/clippy_utils/src/sym_helper.rs
At one time I considered using the `sym` macro here, but after seeing [this discussion](https://rust-lang.zulipchat.com/#narrow/stream/257328-clippy/topic/Checking.20a.20Symbol.20is.20equal.20to.20a.20string.20literal) on Zulip, I thought it would be better to convert it to a string and compare.

>  I think it's fine for us to just universally have this lint?

OK, I'll make change.

Reply:
Ah, I mistakenly assumed that the above discussion was about method name comparison (`iter` or `any`). Since it is a comparison with the type of a slice, I could have simply used the `match` macro, so I fixed it: https://github.com/rust-lang/rust-clippy/pull/13817/commits/86fee9ad5d7be72d41966bf40cdd6380dad58b88